### ISSUE TITLE
KeyError when using footnotes in nested blocks

### ISSUE THREAD

**Issue by frol** [brechtm/rinohtype#95]
I am using Sphinx and here is a piece of code which breaks Rinoh:

```rst
.. note::
    
    Here is my footnote [#qq]_.

    .. [$qq] Here it is!
```

Building this documentation with Rinoh, I am getting the following KeyError exception:

```
# Sphinx version: 1.5
# Python version: 3.5.2 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
#   install_codemeter_on_linux
#   install_codemeter_on_windows
#   getting_started
#   data
#   treenet
#   gps
#   engine
#   
#   resolving references...
#   rendering...
# Loaded extensions:
#   sphinx.domains.python (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/python.py
#   sphinx.builders.text (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/text.py
#   sphinx.ext.imgmath (1.5) from ~/miniconda/lib/python3.5/site-packages/sphinx/ext/imgmath.py
#   sphinx.roles (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/roles.py
#   sphinx.builders.manpage (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/manpage.py
#   sphinx.domains.cpp (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/cpp.py
#   sphinx.ext.autodoc (1.5) from ~/miniconda/lib/python3.5/site-packages/sphinx/ext/autodoc.py
#   sphinx.builders.linkcheck (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/linkcheck.py
#   sphinx.builders.epub (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/epub.py
#   sphinx.builders.epub3 (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/epub3.py
#   sphinx.directives.other (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/directives/other.py
#   alabaster (0.7.9) from ~/miniconda/lib/python3.5/site-packages/alabaster/__init__.py
#   rinoh.frontend.sphinx (unknown version) from ~/miniconda/lib/python3.5/site-packages/rinoh/frontend/sphinx/__init__.py
#   sphinx.builders.xml (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/xml.py
#   sphinx.domains.rst (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/rst.py
#   sphinx.builders.htmlhelp (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/htmlhelp.py
#   sphinx.directives.code (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/directives/code.py
#   sphinx.builders.html (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/html.py
#   sphinx.ext.napoleon (1.5) from ~/miniconda/lib/python3.5/site-packages/sphinx/ext/napoleon/__init__.py
#   sphinx.builders.applehelp (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/applehelp.py
#   sphinx.ext.coverage (1.5) from ~/miniconda/lib/python3.5/site-packages/sphinx/ext/coverage.py
#   sphinx.builders.dummy (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/dummy.py
#   sphinx.domains.c (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/c.py
#   sphinx.builders.devhelp (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/devhelp.py
#   sphinx.builders.texinfo (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/texinfo.py
#   sphinx.builders.latex (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/latex.py
#   sphinx.directives (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/directives/__init__.py
#   sphinx.builders.gettext (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/gettext.py
#   sphinx.builders.websupport (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/websupport.py
#   sphinx.builders.qthelp (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/qthelp.py
#   sphinx.domains.std (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/std.py
#   sphinx.builders.changes (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/builders/changes.py
#   sphinx.directives.patches (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/directives/patches.py
#   sphinx.domains.javascript (unknown version) from ~/miniconda/lib/python3.5/site-packages/sphinx/domains/javascript.py
Traceback (most recent call last):
  File "~/miniconda/lib/python3.5/site-packages/sphinx/cmdline.py", line 296, in main
    app.build(opts.force_all, filenames)
  File "~/miniconda/lib/python3.5/site-packages/sphinx/application.py", line 333, in build
    self.builder.build_update()
  File "~/miniconda/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 246, in build_update
    self.build(['__all__'], to_build)
  File "~/miniconda/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 322, in build
    self.write(docnames, list(updated_docnames), method)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/frontend/sphinx/__init__.py", line 220, in write
    targetname)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/frontend/sphinx/__init__.py", line 242, in write_doc
    rinoh_document.render(outfilename)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/document.py", line 432, in render
    self.part_page_counts = self._render_pages()
  File "~/miniconda/lib/python3.5/site-packages/rinoh/document.py", line 496, in _render_pages
    part_page_count += part.render(part_page_count.count + 1)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/document.py", line 229, in render
    page.render()
  File "~/miniconda/lib/python3.5/site-packages/rinoh/document.py", line 160, in render
    super().render(CONTENT, rerender=index > 0)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/layout.py", line 185, in render
    child.render(type, rerender)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/layout.py", line 185, in render
    child.render(type, rerender)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/layout.py", line 273, in render
    self._render(type, rerender)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/layout.py", line 327, in _render
    self.chain.render(self, rerender=rerender)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/layout.py", line 616, in render
    self._state.flowable_state)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 394, in render
    **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 421, in _flow_with_next
    state=state.first_flowable_state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 394, in render
    **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 421, in _flow_with_next
    state=state.first_flowable_state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 394, in render
    **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 421, in _flow_with_next
    state=state.first_flowable_state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 807, in flow
    return super().flow(container, last_descender, state)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 394, in render
    **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 421, in _flow_with_next
    state=state.first_flowable_state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 394, in render
    **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 421, in _flow_with_next
    state=state.first_flowable_state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 155, in flow
    state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/flowable.py", line 191, in flow_inner
    self.render(pad_cntnr, descender, state=state, **kwargs)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/paragraph.py", line 666, in render
    typeset_line(line, last_line=True)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/paragraph.py", line 632, in typeset_line
    descender, last_line, force)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/paragraph.py", line 934, in typeset
    glyph_span.span.before_placing(container)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/style.py", line 586, in before_placing
    self.parent.before_placing(container)
  File "~/miniconda/lib/python3.5/site-packages/rinoh/reference.py", line 223, in before_placing
    note = container.document.elements[self.target_id(container.document)]
KeyError: '%getting_started#qq'
```

**Comment by brechtm** [comment]
I suppose you mean (`#` instead of `$`):

```rst
.. note::
    
    Here is my footnote [#qq]_.

    .. [#qq] Here it is!
```

**Comment by frol** [comment]
Yeah, I think that was a result of experiments...

**Comment by brechtm** [comment]
I already fixed one bug that this example uncovered. I'm now working the second one.

**Comment by brechtm** [comment]
Nope, it was just that one bug. The fix will be included in 0.3.2.
